### ISSUE TITLE
1.2 release preparations

### ISSUE THREAD

**Issue by akarnokd** [ReactiveX/RxJava#4557]
### Version 1.2.0 - September 17, 2016 ([Maven](http://search.maven.org/#artifactdetails%7Cio.reactivex%7Crxjava%7C1.2.0%7C))

This is a minor release that is functionally equivalent to 1.1.10 minus the removal of some deprecated experimental APIs.
#### Promote `@Beta` to standard (`@since 1.2`)
- in `rx.Observable`
  - `create(SyncOnSubscribe<S, T>)`
  - `doOnRequest(Action1<Long>)`
  - `flatMap(Func1<? super T, ? extends Observable<? extends R>>, Func1<? super Throwable, ? extends Observable<? extends R>>, Func0<? extends Observable<? extends R>>, int)`
  - `flatMap(Func1<? super T, ? extends Observable<? extends R>>, int)`
  - `flatMap(Func1<? super T, ? extends Observable<? extends U>>, Func2<? super T, ? super U, ? extends R>, int)`
  - `flatMapIterable(Func1<? super T, ? extends Iterable<? extends R>>, int)
    rx.Observable.flatMapIterable(Func1<? super T, ? extends Iterable<? extends U>>, Func2<? super T, ? super U, ? extends R>, int)`
  - `fromCallable(Callable<? extends T>)`
  - `toSingle()`
- `rx.Single` (the class itself)
  - `fromCallable(Callable<? extends T>)`
- `rx.SingleSubscriber<T>`
- in `rx.observables.ConnectableObservable`
  - `autoConnect()`
  - `autoConnect(int, Action1<? super Subscription>)`
  - `autoConnect(int)`
- `rx.observables.SyncOnSubscribe<S, T>`
- in `rx.subjects.AsyncSubject`
  - `getThrowable()`
  - `getValue()`
  - `hasCompleted()`
  - `hasThrowable()`
  - `hasValue()`
- in `rx.subjects.BehaviorSubject`
  - `getThrowable()`
  - `getValue()`
  - `getValues()`
  - `getValues(T[])`
  - `hasCompleted()`
  - `hasThrowable()`
  - `hasValue()`
- in `rx.subjects.PublishSubject`
  - `getThrowable()`
  - `hasCompleted()`
  - `hasThrowable()`
- in `rx.subjects.ReplaySubject`
  - `getThrowable()`
  - `getValue()`
  - `getValues()`
  - `getValues(T[])`
  - `hasAnyValue()`
  - `hasCompleted()`
  - `hasThrowable()`
  - `hasValue()`
  - `size()`
#### Promote `@Experimental` to `@Beta`
- `rx.BackpressureOverflow`
- in `rx.Observable`
  - `concatDelayError(Iterable<? extends Observable<? extends T>>)`
  - `concatDelayError(Observable<? extends Observable<? extends T>>)`
  - `concatEager(Iterable<? extends Observable<? extends T>>, int)`
  - `concatEager(Iterable<? extends Observable<? extends T>>)`
  - `concatEager(Observable<? extends Observable<? extends T>>, int)`
  - `concatEager(Observable<? extends Observable<? extends T>>)`
  - `concatEager(Observable<? extends T>, Observable<? extends T>, Observable<? extends T>, Observable<? extends T>, Observable<? extends T>, Observable<? extends T>, Observable<? extends T>, Observable<? extends T>, Observable<? extends T>)`
  - `concatEager(Observable<? extends T>, Observable<? extends T>, Observable<? extends T>, Observable<? extends T>, Observable<? extends T>, Observable<? extends T>, Observable<? extends T>, Observable<? extends T>)`
  - `concatEager(Observable<? extends T>, Observable<? extends T>, Observable<? extends T>, Observable<? extends T>, Observable<? extends T>, Observable<? extends T>, Observable<? extends T>)`
  - `concatEager(Observable<? extends T>, Observable<? extends T>, Observable<? extends T>, Observable<? extends T>, Observable<? extends T>, Observable<? extends T>)`
  - `concatEager(Observable<? extends T>, Observable<? extends T>, Observable<? extends T>, Observable<? extends T>, Observable<? extends T>)`
  - `concatEager(Observable<? extends T>, Observable<? extends T>, Observable<? extends T>, Observable<? extends T>)`
  - `concatEager(Observable<? extends T>, Observable<? extends T>, Observable<? extends T>)`
  - `concatEager(Observable<? extends T>, Observable<? extends T>)`
  - `concatMapDelayError(Func1<? super T, ? extends Observable<? extends R>>)`
  - `concatMapEager(Func1<? super T, ? extends Observable<? extends R>>, int, int)`
  - `concatMapEager(Func1<? super T, ? extends Observable<? extends R>>, int)`
  - `concatMapEager(Func1<? super T, ? extends Observable<? extends R>>)`
  - `delaySubscription(Observable<U>)`
  - `distinctUntilChanged(Func2<? super T, ? super T, Boolean>)`
  - `mergeDelayError(Observable<? extends Observable<? extends T>>, int)`
  - `onBackpressureBuffer(long, Action0, Strategy)`
  - `switchMapDelayError(Func1<? super T, ? extends Observable<? extends R>>)`
  - `switchOnNextDelayError(Observable<? extends Observable<? extends T>>)`
  - `toCompletable()`
  - `toSortedList(Func2<? super T, ? super T, Integer>, int)`
  - `toSortedList(int)`
  - `using(Func0<Resource>, Func1<? super Resource, ? extends Observable<? extends T>>, Action1<? super Resource>, boolean)`
- in `rx.observables.BlockingObservable`
  - `subscribe()`
  - `subscribe(Action1<? super T>, Action1<? super Throwable>, Action0)`
  - `subscribe(Action1<? super T>, Action1<? super Throwable>)`
  - `subscribe(Action1<? super T>)`
  - `subscribe(Observer<? super T>)`
  - `subscribe(Subscriber<? super T>)`
- `rx.Completable`
- in `rx.Single`
  - `defer(Callable<Single<T>>)`
  - `delay(long, TimeUnit, Scheduler)`
  - `delay(long, TimeUnit)`
  - `delaySubscription(Observable<?>)`
  - `doAfterTerminate(Action0)`
  - `doOnError(Action1<Throwable>)`
  - `doOnSubscribe(Action0)`
  - `doOnSuccess(Action1<? super T>)`
  - `doOnUnsubscribe(Action0)`
  - `lift(Operator<? extends R, ? super T>)`
  - `onErrorResumeNext(Func1<Throwable, ? extends Single<? extends T>>)`
  - `onErrorResumeNext(Single<? extends T>)`
  - `toBlocking()`
  - `toCompletable()`
  - `using(Func0<Resource>, Func1<? super Resource, ? extends Single<? extends T>>, Action1<? super Resource>, boolean)`
  - `using(Func0<Resource>, Func1<? super Resource, ? extends Single<? extends T>>, Action1<? super Resource>)`
- `rx.exceptions.CompositeException.CompositeException(Throwable...)`
- in `rx.exceptions.Exceptions`
  - `throwOrReport(Throwable, Observer<?>, Object)`
  - `throwOrReport(Throwable, Observer<?>)`
  - `throwOrReport(Throwable, SingleSubscriber<?>)`
  - `rx.singles.BlockingSingle<T>`
#### Removed
- in `rx.Observable`
  - `extend(Func1<OnSubscribe<T>, R>)` : use `to(Func1)` instead
  - `fromAsync()` : renamed to `fromEmitter()`
- in `rx.Completable`
  - `CompletableSubscriber` : now `rx.CompletableSubscriber`
  - `CompletableOnSubscribe` : renamed to `Completable.OnSubscribe`
  - `CompletableOperator` : renamed to `Completable.Operator`
  - `CompletableTransformer` : renamed to `Completable.Transformer`

**Comment by wispborne** [comment]
Does this mean that the `Completable` api is not expected to change before becoming stable? 
Put another way, is it _reasonably_ safe to use `Completable` in a large project without needing to worry about new versions breaking it?

**Comment by akarnokd** [comment]
https://github.com/ReactiveX/RxJava#beta

TLDR; it means it won't be removed in a patch release like 1.2.1 but no guarantees about it in 1.3 - it is unlikely though and unless there is some serious API flaw with it, the existing methods won't change either.

**Comment by akarnokd** [comment]
Closing via release 1.2.0
